plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobStore.java

```diff
@@ -76,6 +79,10 @@ class S3BlobStore implements BlobStore {
     private final AsyncExecutorContainer normalExecutorBuilder;
     private final boolean multipartUploadEnabled;
 
+    final Setting<Boolean> SOFT_DELETE_ENABLED = Setting.boolSetting("soft_delete.enabled", true);
```

Comment:
do we need to add Node.Scope

Reply:
this is related to repo metadata only . So not added it . Same is followed for other repo settings as well .